taiga_halo2/src/merkle_tree.rs

```diff
@@ -9,13 +9,48 @@ use halo2_proofs::arithmetic::Field;
 use pasta_curves::pallas;
 use rand::distributions::{Distribution, Standard};
 use rand::{Rng, RngCore};
+#[cfg(feature = "nif")]
+use rustler::NifTuple;
+use subtle::CtOption;
 
 #[cfg(feature = "serde")]
 use serde;
 
 #[cfg(feature = "borsh")]
 use borsh::{BorshDeserialize, BorshSerialize};
 
+/// The root of the note commitment Merkletree.
+#[derive(Eq, PartialEq, Clone, Copy, Debug)]
+#[cfg_attr(feature = "nif", derive(NifTuple))]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct Anchor(pallas::Base);
+
+impl Anchor {
```

Comment:
For now, the `Anchor` is exactly the root. I'm considering extending the `Anchor` struct. We probably will add more parameters to `Anchor`, like the block number, epoch, etc., so that we can add constraints in circuits and check them publicly. We discussed the proposal before but have yet to decide what to add.
Besides, `Node` is a little confusing; we also have the blockchain Node. I don't tend to use the Node struct outside of the Merkle tree and may make it private later. 
Does it make any sense? If not, I can revert it.

Reply:
Yeah, it does. However, the executor does work with Node objects too when constructing the tree.